jdaviz/configs/imviz/tests/test_orientation.py

```diff
@@ -152,3 +156,48 @@ def test_custom_orientation(self):
         lc_plugin._obj.add_orientation(rotation_angle=None, east_left=True, label=None,
                                        set_on_create=True, wrt_data=None)
         assert self.viewer.state.reference_data.label == "CCW 42.00 deg (E-left)"
+
+
+class TestDeleteOrientation(BaseImviz_WCS_WCS):
+
+    @pytest.mark.parametrize("klass", [EllipseSkyRegion, RectangleSkyRegion])
+    @pytest.mark.parametrize(
+        ("angle", "sbst_theta"),
+        [(0.5 * u.rad, 2.641593),
+         (-0.5 * u.rad, 3.641589)])
+    def test_delete_orientation_with_subset(self, klass, angle, sbst_theta):
+        lc_plugin = self.imviz.plugins['Orientation']
+        lc_plugin.link_type = 'WCS'
+
+        # Should automatically be applied as reference to first viewer.
+        lc_plugin._obj.create_north_up_east_left(set_on_create=True)
+
+        # Create rotated shape
+        reg = klass(center=SkyCoord(ra=337.51931488, dec=-20.83187472, unit="deg"),

```

Comment:
Providing region as input made more sense to me, so I did that change here. Also added test case for rectangle that seems to have a separate logic route. Luckily ellipse and rectangle have same inputs in regions, so we can easily parametrize them.